src/room17/SkyBlock/SkyBlockListener.php

```diff
@@ -236,5 +226,10 @@ public function onQuit(PlayerQuitEvent $event): void {
             $isle->tryToClose();
         }
     }
-
-}
\ No newline at end of file
+    /**
+     * @param PlayerQuitEvent $event
+     */
+    public function onSleep(PlayerBedEnterEvent $event){
```

Comment:
It's not follow SkyBlock's code-style, also, you have to check that the player is in an island before cancelling the event.